File: problems/169. Majority Element/Solution.java
class Solution {
    public int majorityElement(int[] nums) {
        if (nums.length == 1) return nums[0];
        int ans = 0, count = 0;
        for (int i = 0; i < nums.length; i++) {
            if (count == 0) ans = nums[i];
            if (nums[i] == ans) count += 1;
            else count -= 1;
        }
        return ans;
    }
}

File: problems/169. Majority Element/Solution.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int ans = 0, count = 0;
        for (const int& num : nums) {
            if (count == 0) ans = num;
            if (num == ans) count++;
            else count--;
        }
        return ans;
    }
};